src/blocks/content-helper/current-post-details/component.tsx

```diff
@@ -0,0 +1,359 @@
+ */
+function TopReferrersSection( data: PostPerformanceData ) {
+	let totalViewsPercentage = 0;
+	return (
+		<div className="section top-referrers">
+			<table>
+				<thead>
+					<tr>
+						<th scope="col">{ __( 'Top Referrers', 'wp-parsely' ) }</th>
+						<th colSpan={ 2 } scope="colgroup">{ __( 'Page Views', 'wp-parsely' ) }</th>
+					</tr>
+				</thead>
+				<tbody>{
+					Object.entries( data.referrers.top ).map( ( [ key, value ] ) => {
+						if ( key === 'totals' ) {
+							totalViewsPercentage = Number( value.viewsPercentage );
+							return null;
+						}
+
+						let referrerUrl = key;
+						if ( key === 'direct' ) {
+							referrerUrl = __( 'Direct', 'wp-parsely' );
+						}
+
+						/* translators: %s: Percentage value, %%: Escaped percent sign */
+						const ariaLabel = sprintf( __( '%s%%', 'wp-parsely' ), value.viewsPercentage ); // eslint-disable-line @wordpress/valid-sprintf
+
+						return (
+							<tr key={ key }>
+								<th scope="row">{ referrerUrl }</th>
+								<td>
+									<div aria-label={ ariaLabel }
+										className="percentage-bar"
+										style={ { '--bar-fill': value.viewsPercentage + '%' } as React.CSSProperties }>
+									</div>
+								</td>
+								<td>{ ImpreciseNumber( value.views ) }</td>
+							</tr>
+						);
+					} )
+				}
+				</tbody>
+			</table>
+			<div> {
+				/* translators: %s: Percentage value, %%: Escaped percent sign */
+				sprintf( _n( // eslint-disable-line @wordpress/valid-sprintf
+					'%s%% of views came from top referrers.',
+					'%s%% of views came from top referrers.',
+					totalViewsPercentage, 'wp-parsely' ), totalViewsPercentage
+				)
+			}
+			</div>
+		</div>
+	);
+}
+
+/**
+ * Outputs the "Actions" section.
+ *
+ * @param {PostPerformanceData} data The data needed to populate the section.
+ */
+function ActionsSection( data: PostPerformanceData ) {
+	const ariaOpensNewTab = <span className="screen-reader-text"> {
+		__( '(opens in new tab)', 'wp-parsely' ) }
+	</span>;
+
+	return (
+		<div className="section actions">
+			<Button
+				href={ data.url } rel="noopener" target="_blank" variant="secondary">
+				{ __( 'Visit Post', 'wp-parsely' ) }{ ariaOpensNewTab }
+			</Button>
+			<Button
+				href={ data.statsUrl } rel="noopener" target="_blank" variant="primary">
+				{ __( 'View in Parse.ly', 'wp-parsely' ) }{ ariaOpensNewTab }
+			</Button>
+		</div>
+	);
+}
+
+/**
+ * Implements the "Imprecise Number" functionality of the Parse.ly dashboard.
+ *
+ * Note: This function is not made to process float numbers.
+ *
+ * @param {string} value          The number to process. It can be formatted.
+ * @param {number} fractionDigits The number of desired fraction digits.
+ * @param {string} glue           A string to put between the number and unit.
+ * @return {string} The number formatted as an imprecise number.
+ */
+function ImpreciseNumber( value: string, fractionDigits = 1, glue = '' ): string {
```

Comment:
PascalCase should be reserved for classes and React components, use camelCase
```suggestion
function impreciseNumber( value: string, fractionDigits = 1, glue = '' ): string {
```